spec/defines/sensu_write_json_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'spec_helper'
+
+CFG_PATH = '/etc/sensu/conf.d/custom.json'
+CUSTOM_CFG = {"custom" => "data"}
+
+describe 'sensu::write_json', :type => :define do
+  let(:facts) { { :osfamily => 'RedHat' } }
+
+  context 'with various params' do
+    let(:title) { CFG_PATH }
+
+    context 'defaults' do
+      let(:params) { {
+        :owner => 'sensu',
+        :group => 'sensu',
+        :content => CUSTOM_CFG,
+      } }
+
+      it { should contain_sensu__write_json(CFG_PATH).with(
+        :ensure => 'present',
+        :mode => '0755',
+        :owner => 'sensu',
+        :group => 'sensu',
+        :pretty => true,
+        :content => CUSTOM_CFG,
+      ) }
+
+      it { should contain_file(CFG_PATH).with(
+        :ensure => 'present',
+        :mode => '0755',
+        :owner => 'sensu',
+        :group => 'sensu',
+        :content => "{\n    \"custom\": \"data\"\n}\n",
+        :notify => nil,
+        :subscribe => nil,
+      ) }
+
+    end
+
+    context 'setting absent' do
+      let(:params) { {
+        :ensure => 'absent',
+        :mode => '0700',
+        :owner => 'root',
+        :group => 'root',
+      } }
+
+      it { should contain_file(CFG_PATH).with(
+        :ensure => 'absent',
+        :mode => '0700',
+        :notify => nil,
+        :subscribe => nil,
+      ) }
+    end
+
+    context 'disable pretty print' do
+      let(:params) { {
+        :owner => 'root',
+        :group => 'root',
+        :pretty => false,
+        :content => CUSTOM_CFG,
+      } }
+
+      it { should contain_file(CFG_PATH).with(
+        :ensure => 'present',
+        :mode => '0755',
+        :owner => 'root',
+        :group => 'root',
+        :content => "{\"custom\":\"data\"}",
```

Comment:
Here's a way to do the HEREDOC so that you dont have all the escaping.

https://github.com/ghoneycutt/puppet-module-bind/blob/master/spec/defines/acl_spec.rb#L22-L35